src/Adapters.Tests/TraceListener.Tests.Shared/ApplicationInsightsTraceListenerTests.cs

```diff
@@ -367,6 +367,25 @@ public void TraceEventDoesNotStoreCallStackInTelemetryPropertiesBecauseLongValue
             Assert.IsFalse(telemetry.Properties.ContainsKey("CallStack"));
         }
 
+        [TestMethod]
+        public void TraceListenerFlushesChannel()
+        {
+            TelemetryConfiguration.Active.TelemetryChannel = this.adapterHelper.Channel;
+
+            using (var listener = new ApplicationInsightsTraceListener(Guid.NewGuid().ToString()))
+            {
+                try
+                {
+                    listener.Flush();
+                    Assert.Fail();
+                }
+                catch (Exception ex)
+                {
+                    Assert.AreEqual("Flush called", ex.Message);
```

Comment:
This looks like rather unconventional way to test Flush(), something with `Assert.AreEqual(true, this.adapterHelper.Channel.Flushed)` will be more in line with the current approach.